crates/persistence/src/thesaurus.rs

```diff
@@ -0,0 +1,38 @@
+use crate::Result;
+use async_trait::async_trait;
+use terraphim_types::Thesaurus;
+
+use crate::Persistable;
+
+#[async_trait]
+impl Persistable for Thesaurus {
+    fn new() -> Self {
+        Thesaurus::new()
+    }
+
+    /// Save to a single profile
+    async fn save_to_one(&self, profile_name: &str) -> Result<()> {
+        self.save_to_profile(profile_name).await?;
+        Ok(())
+    }
+
+    // Saves to all profiles
+    async fn save(&self) -> Result<()> {
+        let _op = &self.load_config().await?.1;
+        let _ = self.save_to_all().await?;
+        Ok(())
+    }
+
+    /// Load key from the fastest operator
+    async fn load(&mut self, key: &str) -> Result<Self> {
+        let op = &self.load_config().await?.1;
+        let obj = self.load_from_operator(key, op).await?;
+        Ok(obj)
+    }
+
+    /// returns ulid as key + .json
+    fn get_key(&self) -> String {
+        // TODO: What should the key be?
+        "thesaurus.json".to_string()
```

Comment:
@AlexMikhalev, any idea for what we could use as a key here?

Reply:
"role shortname"+"thesaurus.json".to_string()
there is always a default role in a config

Reply:
We don't have the roles in the `Thesaurus`.

Reply:
I was thinking about passing the role to the constructor, like `Thesaurus::new("role")`, but the `new` method in `Persistable` trait doesn't take any arguments.

Reply:
Check out https://github.com/terraphim/terraphim-ai/blob/main/crates/config/src/lib.rs - now has a default, which can be the default prefix, as in the default role.

Reply:
Good idea. Done.